src/RandomEnchants.cpp

```diff
@@ -1,63 +1,139 @@
+		Looted = sConfigMgr->GetBoolDefault("RandomEnchants.OnLoot", 1);
+		QuestReward = sConfigMgr->GetBoolDefault("RandomEnchants.OnQuestReward", 1);
+		HighQuality = sConfigMgr->GetIntDefault("RandomEnchants.HighQuality", 1);
+		LowQuality = sConfigMgr->GetIntDefault("RandomEnchants.LowQuality", 1);
+		Chance1 = sConfigMgr->GetIntDefault("RandomEnchants.Chance1", 60);
+		Chance2 = sConfigMgr->GetIntDefault("RandomEnchants.Chance2", 65);
+		Chance3 = sConfigMgr->GetIntDefault("RandomEnchants.Chance3", 70);
+
+		// Sanitize
+		if (HighQuality > 5) { HighQuality = 5; }
+		
+	}
+};
+
+class RandomEnchantsPlayer : public PlayerScript {
+public:
+
+	RandomEnchantsPlayer() : PlayerScript("RandomEnchantsPlayer") { }
+
+	void OnLogin(Player* player)  override
+	{
+		if (sConfigMgr->GetBoolDefault("RandomEnchants.Announce", true)) {
+			ChatHandler(player->GetSession()).SendSysMessage("This server is running the |cff4CFF00RandomEnchants |rmodule.");
+		}
+	}
 
-    void OnLogin(Player* player) override {
-		if (sConfigMgr->GetBoolDefault("RandomEnchants.AnnounceOnLogin", true))
-            ChatHandler(player->GetSession()).SendSysMessage(sConfigMgr->GetStringDefault("RandomEnchants.OnLoginMessage", "This server is running a RandomEnchants Module.").c_str());
-    }
 	void OnLootItem(Player* player, Item* item, uint32 /*count*/, uint64 /*lootguid*/) override
 	{
-		if (sConfigMgr->GetBoolDefault("RandomEnchants.OnLoot", true))
+		if (RandomEnchantEnabled)
+		{
 			RollPossibleEnchant(player, item);
+		}
 	}
+
 	void OnCreateItem(Player* player, Item* item, uint32 /*count*/) override
 	{
-		if (sConfigMgr->GetBoolDefault("RandomEnchants.OnCreate", true))
-			RollPossibleEnchant(player, item);
+		if (RandomEnchantEnabled)
+		{
+			if (Crafted == true)
+			{
+				RollPossibleEnchant(player, item);
+			}
+		}
 	}
+
 	void OnQuestRewardItem(Player* player, Item* item, uint32 /*count*/) override
 	{
-		if(sConfigMgr->GetBoolDefault("RandomEnchants.OnQuestReward", true))
-			RollPossibleEnchant(player, item);
+		if (RandomEnchantEnabled)
+		{
+			if (QuestReward == true)
+			{
+				RollPossibleEnchant(player, item);
+			}
+		}
 	}
+
 	void RollPossibleEnchant(Player* player, Item* item)
 	{
 		uint32 Quality = item->GetTemplate()->Quality;
 		uint32 Class = item->GetTemplate()->Class;
-		if ((Quality > 5 && Quality < 1)/*eliminates enchanting anything that isn't a recognized quality*/ || (Class != 2 && Class != 4)/*eliminates enchanting anything but weapons/armor*/)
+
+		// Weed out items ( needs more testing )
+		if ((Quality < LowQuality && Quality > HighQuality) || (Class != 2 && Class != 4))
+		{
+			// Eliminate enchanting anything that isn't a designated quality.
+			// Eliminate enchanting anything but weapons(2) and armor(4).
 			return;
+		}
+
+		// It's what we want..
 		int slotRand[3] = { -1, -1, -1 };
 		uint32 slotEnch[3] = { 0, 1, 5 };
 		double roll1 = rand_chance();
-		if (roll1 >= 70.0)
+		if (roll1 >= Chance1) //
```

Comment:
```suggestion
		if (roll1 >= Chance1)
```

? I remove the // dunno what they're for